libglusterfs/src/inode.c

```diff
@@ -719,10 +719,10 @@ inode_new(inode_table_t *table)
             __inode_ref(inode, false);
         }
         pthread_mutex_unlock(&table->lock);
-    }
 
-    /* let the dummy, 'unlinked' inodes have root as namespace */
-    inode->ns_inode = inode_ref(table->root);
+        /* let the dummy, 'unlinked' inodes have root as namespace */
+        inode->ns_inode = inode_ref(table->root);
```

Comment:
Perhaps instead set it after line 719, as:
```
inode->ns_inode = __inode_ref(table->root, false);
```

So you can save one cycle of lock/unlock of table->lock ?

Reply:
Yes, it can save.
But I think it will be cleaner to apply it on a different PR as this new prototype (w/o an option to default value) will lead to many tiny changes. @amarts - what do u think?

Reply:
Of course, could be done in a different patch.

Reply:
opened https://github.com/gluster/glusterfs/issues/3034